hearth/lib/hearth/middleware/retry.rb

```diff
@@ -2,114 +2,93 @@
 
 module Hearth
   module Middleware
-    # A middleware that retries the request.
+    # A middleware that retries the request using a retry strategy.
     # @api private
     class Retry
-      # Max backoff (in seconds)
-      MAX_BACKOFF = 20
-
+      # TODO: explain RetryStrategy class in docs or make an interface
       # @param [Class] app The next middleware in the stack.
-      # @param [Boolean] retry_mode Specifies which retry algorithm to use.
-      #   Values are:
-      #   * `standard` - A standardized set of retry rules across the AWS SDKs.
-      #     This includes support for retry quotas, which limit the number of
-      #     unsuccessful retries a client can make.
-      #   * `adaptive` - An experimental retry mode that includes all the
-      #     functionality of `standard` mode along with automatic client side
-      #     throttling.  This is a provisional mode that may change behavior
-      #     in the future.
-      # @param [String] max_attempts An integer representing the maximum number
-      #   of attempts that will be made for a single request, including the
-      #   initial attempt.
-      # @param [Boolean] adaptive_retry_wait_to_fill Used only in `adaptive`
-      #   retry mode. When true, the request will sleep until there is
-      #   sufficient client side capacity to retry the request. When false,
-      #   the request will raise a `CapacityNotAvailableError` and will not
-      #   retry instead of sleeping.
-      def initialize(app, retry_mode:, max_attempts:,
-                     adaptive_retry_wait_to_fill:, error_inspector_class:,
-                     retry_quota:, client_rate_limiter:)
+      # @param [RetryStrategy] retry_strategy The retry strategy to use.
```

Comment:
Do we care to make an interface with not implemented errors?

Reply:
Since there is no shared code and its ruby idiomatic to not have an "interface" class, lets skip it for now.